values/src/dimension.rs

```diff
@@ -490,44 +491,170 @@ pub enum Ambiguity {
+            Form::YearMonthDay(_) => Some(true),
+            Form::TimeOfDay(_) => Some(false),
+            Form::DayOfWeek { .. } => Some(true),
+            Form::Empty => None,
+            Form::PartOfDay { .. } => Some(false),
+            Form::Meal => Some(false),
+            Form::Celebration => Some(true),
+            Form::PartOfMonth => Some(true),
+            Form::PartOfYear => Some(true),
+            Form::DayOfMonth => Some(true),
+            Form::PartOfForm(_) => None,
+            Form::PartOfWeek => Some(true),
+            Form::Span => None,
+        }
+    }
+
+    pub fn time_form(&self) -> Option<bool> {
+        match self.form {
+            Form::Cycle(grain) => {
+                match grain {
+                    Grain::Hour => Some(true),
+                    Grain::Minute => Some(true),
+                    Grain::Second => Some(true),
+                    _ => Some(false),
+                }
+            },
+            Form::Year(_) => Some(false),
+            Form::Month(_) => Some(false),
+            Form::MonthDay(_) => Some(false),
+            Form::YearMonthDay(_) => Some(false),
+            Form::TimeOfDay(_) => Some(true),
+            Form::DayOfWeek { .. } => Some(false),
+            Form::Empty => None,
+            Form::PartOfDay { .. } => Some(true),
+            Form::Meal => Some(true),
+            Form::Celebration => Some(false),
+            Form::PartOfMonth => Some(false),
+            Form::PartOfYear => Some(false),
+            Form::DayOfMonth => Some(false),
+            Form::PartOfForm(_) => None,
+            Form::PartOfWeek => Some(false),
+            Form::Span => None,
+        }
+    }
+
+    pub fn period_form(&self) -> Option<bool> {
+        match self.form {
+            Form::Cycle(grain) => {
+                match grain {
+                    Grain::Year => Some(true),
+                    Grain::Quarter => Some(true),
+                    Grain::Month => Some(true),
+                    Grain::Week => Some(true),
+                    _ => Some(false),
+
+                }
+            },
+            Form::Year(_) => Some(true),
+            Form::Month(_) => Some(true),
+            Form::MonthDay(_) => Some(false),
+            Form::YearMonthDay(_) => Some(false),
+            Form::TimeOfDay(_) => Some(false),
+            Form::DayOfWeek { .. } => Some(false),
+            Form::Empty => Some(false),
+            Form::PartOfDay { .. } => Some(true),
+            Form::Meal => Some(true),
+            Form::Celebration => Some(false),
+            Form::PartOfMonth => Some(true),
+            Form::PartOfYear => Some(true),
+            Form::DayOfMonth => Some(false),
+            Form::PartOfForm(_) => None,
+            Form::PartOfWeek => Some(true),
+            Form::Span => Some(true),
+        }
+    }
+
+    pub fn set_datetime_kind(&mut self, datetime_kind: DatetimeKind) {
+        self.datetime_kind = datetime_kind
+    }
```

Comment:
I don't think this method is needed as the `datetime_kind` property is public.

Reply:
✅